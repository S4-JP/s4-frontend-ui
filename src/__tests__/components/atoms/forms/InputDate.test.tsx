import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import { InputDate } from '../../../../index'

describe('form - InputDate のレンダリング確認', () => {
  test('オプションなしで設定した場合、初期値でレンダリングされる。', () => {
    render(<InputDate value={null} />)

    expect(screen.getByTestId('atoms-f-input-date')).toMatchSnapshot()
  })

  test('propsで値を渡した場合、その値でレンダリングされる。', () => {
    render(
      <InputDate
        name="test date-picker"
        placeholder="Please select a date"
        minDate="2022/01/01"
        value="2022/02/22"
        onChange={jest.fn()}
      />
    )

    expect(screen.getByTestId('atoms-f-input-date')).toMatchSnapshot()
  })

  test('入力して日付を変更した場合、入力された日付のdate object でコールされる。', () => {
    const mockOnChange = jest.fn()
    render(
      <InputDate
        name="test date-picker"
        placeholder="Please select a date"
        minDate="2022/01/01"
        value="2022/02/22"
        onChange={mockOnChange}
      />
    )

    fireEvent.change(screen.getByRole('textbox'), {
      target: { value: '2022-06-15' },
    })
    expect(mockOnChange).toBeCalledTimes(1)
    expect(mockOnChange).toHaveBeenCalledWith(new Date('2022-06-15'))
  })
})
