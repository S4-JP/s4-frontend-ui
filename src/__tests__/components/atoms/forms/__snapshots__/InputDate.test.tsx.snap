// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form - InputDate のレンダリング確認 propsで値を渡した場合、その値でレンダリングされる。 1`] = `
<div
  class="sc-ksZaOG cBAAyC"
  data-testid="atoms-f-input-date"
>
  <div
    class="sc-hAZoDl bbaXIT"
  >
    <div
      class="sc-fnykZs kgybnE"
    >
      <div
        class="react-datepicker-wrapper"
      >
        <div
          class="react-datepicker__input-container"
        >
          <input
            autocomplete="off"
            class=""
            name="test date-picker"
            placeholder="Please select a date"
            type="text"
            value="2022/02/22"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`form - InputDate のレンダリング確認 オプションなしで設定した場合、初期値でレンダリングされる。 1`] = `
<div
  class="sc-ksZaOG cBAAyC"
  data-testid="atoms-f-input-date"
>
  <div
    class="sc-hAZoDl bbaXIT"
  >
    <div
      class="sc-fnykZs kgybnE"
    >
      <div
        class="react-datepicker-wrapper"
      >
        <div
          class="react-datepicker__input-container"
        >
          <input
            autocomplete="off"
            class=""
            placeholder="0000/00/00"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
